File: src/zmq_client.cpp
#include <iostream>
#include <string>

#include <zmq.hpp>

int main(int argc, char *argv[]) {
    zmq::context_t context(1);
    zmq::socket_t socket(context, zmq::socket_type::req);
    socket.connect("tcp://localhost:5555");

    std::cout << "Connected to server\n";

    // コマンドライン引数からメッセージを取得、なければ "Hello"
    std::string msg = (argc > 1) ? argv[1] : "Hello";

    // メッセージ送信
    socket.send(zmq::buffer(msg), zmq::send_flags::none);
    std::cout << "Sent: " << msg << "\n";

    // 応答受信
    zmq::message_t reply;
    auto result = socket.recv(reply, zmq::recv_flags::none);
    if (!result) {
        std::cerr << "Failed to receive reply\n";
        return 1;
    }
    std::string reply_str(static_cast<char *>(reply.data()), reply.size());
    std::cout << "Received: " << reply_str << "\n";

    return 0;
}

File: python/zmq_client.py
#!/usr/bin/env python3
"""ZeroMQ client example in Python."""

import sys
import zmq


def run():
    """Run the ZeroMQ client."""
    # サーバーアドレス
    server_address = 'tcp://localhost:5555'

    # コマンドライン引数からメッセージを取得、なければ "Hello"
    message = sys.argv[1] if len(sys.argv) > 1 else "Hello"

    # ZeroMQコンテキストとソケットを作成
    context = zmq.Context()
    socket = context.socket(zmq.REQ)
    socket.connect(server_address)

    print(f"Connected to server")

    # メッセージ送信
    socket.send_string(message)
    print(f"Sent: {message}")

    # 応答受信
    reply = socket.recv_string()
    print(f"Received: {reply}")


if __name__ == '__main__':
    run()
